Backport of:

From 030c376fe4c97aea2178996e658b4c33aa9a736a Mon Sep 17 00:00:00 2001
From: cristy <cristy@aa41f4f7-0bf4-0310-aa73-e5a19afd5a74>
Date: Tue, 28 Oct 2014 13:11:48 +0000
Subject: [PATCH] Quit earlier in case of corrupted pnm image

Do not try to read a corrupted pnm file. Bail early.

Fix a segv and thus a DOS

git-svn-id: https://subversion.imagemagick.org/subversion/ImageMagick/branches/ImageMagick-6@16867 aa41f4f7-0bf4-0310-aa73-e5a19afd5a74
origin: http://trac.imagemagick.org/changeset/16867
---
 coders/pnm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: imagemagick-6.6.9.7/coders/pnm.c
===================================================================
--- imagemagick-6.6.9.7.orig/coders/pnm.c	2016-11-14 14:11:53.385846295 -0500
+++ imagemagick-6.6.9.7/coders/pnm.c	2016-11-14 14:11:53.381846295 -0500
@@ -402,7 +402,7 @@
       }
     if ((image->columns == 0) || (image->rows == 0))
       ThrowReaderException(CorruptImageError,"NegativeOrZeroImageSize");
-    if (max_value >= 65536)
+    if ((max_value == 0) || (max_value > 65536))
       ThrowReaderException(CorruptImageError,"ImproperImageHeader");
     for (depth=1; GetQuantumRange(depth) < max_value; depth++) ;
     image->depth=depth;
